src/node.c

```diff
@@ -557,6 +574,26 @@ freesasa_node_atom_pdb_line(const freesasa_node *node)
     return node->properties.atom.pdb_line;
 }
 
+const char *
+freesasa_node_atom_residue_number(const freesasa_node *node)
+{
+    assert(node->type == FREESASA_NODE_ATOM);
+    return node->properties.atom.res_number;
+}
+
+const char *
+freesasa_node_atom_residue_name(const freesasa_node *node)
+{
+    assert(node->type == FREESASA_NODE_ATOM);
+    return node->properties.atom.res_name;
+}
+
+char freesasa_node_atom_chain(const freesasa_node *node)
+{
+    assert(node->type == FREESASA_NODE_ATOM);
+    return node->properties.atom.chain;
+}
+
```

Comment:
These three could be implemented by traversing the tree upwards using `freesasa_node_parent()`, that way you don't need to store duplicate copies.

Reply:
Yeah that's how I originally implemented but it didn't work. There was a bug when it came to finding hetatm for the appropriate chain. With this implementation I fixed the bug. Maybe we should look into why the chains don't line up when parsing hetatm?